data-model/src/main/java/io/micronaut/data/model/schema/DbTypes.java

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.model.schema;
+
+import io.micronaut.core.annotation.Internal;
+
+import java.sql.Types;
+
+/**
+ * Provides constants and utility methods for working with database types.
+ * This is basically mapping of {@link java.sql.Types} with some additional field type values
+ * with option to adding new types later if needed.
+ * This class contains constants for various database types, such as numeric,
+ * character, and date/time types. It also provides methods for checking whether
+ * a given type code represents a specific type, such as a numeric or character
+ * type.
+ *
+ * @author radovanradic
+ * @since 4.13.0
+ */
+@Internal
+public final class DbTypes {
```

Comment:
maybe this should be an `enum` with a `type()` method for the `int`

Reply:
Updated